Solution added

Inzilbeth opened the pull request.

Inzilbeth committed:
Solution added

yurii-litvinov commented on Homework1/Task1/MatrixMultiplication/Matrix.cs:
```diff
@@ -0,0 +1,257 @@
+    public class Matrix
+    {
+        public int Rows { get; set; }
+        public int Columns { get; set; }
```
public-сеттер у этих свойств вызывает удивление, а отсутствие комментариев не даёт его развеять. Не уверен, что их вообще стоит отдельно хранить, matrixArray ведь знает свой размер по обоим измерениям.

yurii-litvinov commented on Homework1/Task1/MatrixMultiplication/Matrix.cs:
```diff
@@ -0,0 +1,257 @@
+        /// <returns>The matrix itself now filled with random values.</returns>
+        public Matrix RandomizeValues(int maxValue)
+        {
+            var random = new Random((int)DateTime.Now.Ticks);
```
Он же, кажется, и так инициализируется таймером

yurii-litvinov commented on Homework1/Task1/MatrixMultiplication/Matrix.cs:
```diff
@@ -0,0 +1,257 @@
+            {
+                for (int j = 0; j < Columns; j++)
+                {
+                    Console.Write($"{matrixArray[i, j]} ");
```
В интерполяции строк можно указывать форматные спецификаторы ширины поля, чтобы матрица выглядела посимпатичнее. См. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated

yurii-litvinov commented on Homework1/Task1/MatrixMultiplication/Matrix.cs:
```diff
@@ -0,0 +1,257 @@
+            }
+
+            return result;
+        }
```
Это обсудили, стоит ограничить число потоков, просто поровну поделив строки между ними

yurii-litvinov commented on Homework1/Task1/MatrixMultiplication/Matrix.cs:
```diff
@@ -0,0 +1,257 @@
+                }
+            }
+
+            return true;
```
Если obj не матрица, то true? Даже если null? Мм.

Inzilbeth replied:
А если методу в качестве параметра передать null объект, разве не вызовется ArgumentNullException?

yurii-litvinov commented on Homework1/Task1/MatrixMultiplication/Parser.cs:
```diff
@@ -0,0 +1,76 @@
+        /// <returns>Built matrix.</returns>
+        public static Matrix Parse(string path)
+        {
+            using (var streamReader = new StreamReader(path))
```
Тут using закрывается в самом конце метода, так что лучше было `using var streamReader = new StreamReader(path);` и дальше без фигурных скобок и лишнего отступа

yurii-litvinov commented on Homework1/Task1/MatrixMultiplicationTests/MatrixMultiplicationTests.cs:
```diff
@@ -0,0 +1,94 @@
+{
+    class MatrixMultiplicationTests
+    {
+        private Matrix Left;
+        private Matrix Right;
+        private Matrix ExpectedResult;
```
А поля именуются со строчной, даже в тестах

yurii-litvinov commented on Homework1/Task1/MatrixMultiplicationTests/Tools.cs:
```diff
@@ -0,0 +1,35 @@
+            {
+                streamWriter.Write(text);
+            }
+        }
```
А есть же File.WriteAllText. То же с Read. Такие утилиты придумали до Вас

yurii-litvinov commented on Homework1/Task1/Task1ConsoleApp/Program.cs:
```diff
@@ -0,0 +1,111 @@
+            stopWatch.Start();
+            if (isThreaded)
+            {
+                var res = Matrix.MultiThreadedMultiply(a, b);
```
Компилятор не ругается на неиспользуемую переменную?

yurii-litvinov reviewed (changes requested):
Пока что по ECTS-ной шкале было бы 3 балла из 4, хотя многие замечания не ошибки, а скорее рекомендации.

Inzilbeth committed:
Fixes

- Removed Rows and Columns properties from Matrix class
- Fixed Random instance initialization
- Improved Matrix.WriteToConsole
- Improved Matrix.MultiThreadedMultiply
- fixed Matrix.Equals
- Refactoring in Parser.Parse
- Fixed variable names in MatrixMultiplicationTests
- Removed Tools class, used File.WriteAllText & File.ReadAllText instead
- Removed unused variables in Main

yurii-litvinov reviewed (approved):
Да, теперь всё отлично, 4 балла.

Inzilbeth committed:
Added matrix pre-transposing for cahce-friendliness
